src/System.Management.Automation/namespaces/ProviderBase.cs

```diff
@@ -1862,6 +1862,24 @@ public void WriteInformation(object messageData, string[] tags)
 
                 result.AddOrSetProperty("PSChildName", childName);
                 providerBaseTracer.WriteLine("Attaching {0} = {1}", "PSChildName", childName);
+#if UNIX
+
+                // Add a commonstat structure to file system objects
+                if (ExperimentalFeature.IsEnabled("PSUnixFileStat") && ProviderInfo.ImplementingType == typeof(Microsoft.PowerShell.Commands.FileSystemProvider))
+                {
+                    try
+                    {
+                        // Use LStat because if you get a link, you want the information about the 
+                        // link, not the file.
+                        var commonStat = Platform.Unix.GetLStat(path);
+                        result.AddOrSetProperty("UnixStat", commonStat);
+                    }
+                    catch
```

Comment:
Do you want to catch a specific exception here?

Reply:
no, any problem means that we can't add the property, so don't. I'll add a comment

Reply:
added a comment